src/google/protobuf/compiler/js/js_generator.cc

```diff
@@ -3477,7 +3477,8 @@ void Generator::GenerateFile(const GeneratorOptions& options,
     GenerateExtension(options, printer, *it);
   }
 
-  if (options.import_style == GeneratorOptions::kImportCommonJs) {
+  // if provided is empty, do not export anything
+  if (options.import_style == GeneratorOptions::kImportCommonJs && provided.size()) {
```

Comment:
Just to make it a little bit clearer, could you do `!provided.empty()` instead of `provided.size()`?

Reply:
changed :)